MediaBrowser.XbmcMetadata/Parsers/NfoParserHelpers.cs

```
@@ -0,0 +1,246 @@
+        {
+            string readerName = xmlReader.Name;
+            if (providerIds.TryGetValue(readerName, out string? providerIdValue))
+            {
+                var id = xmlReader.ReadElementContentAsString();
+                if (!string.IsNullOrWhiteSpace(providerIdValue) && !string.IsNullOrWhiteSpace(id))
+                {
+                    item.SetProviderId(providerIdValue, id);
+                }
+            }
+            else
+            {
+                xmlReader.Skip();
+            }
+        }
+
+        internal static string? ReadTrailerUrlFromNfo(this XmlReader reader)
+        {
+            var val = reader.ReadStringFromNfo();
+
+            if (!string.IsNullOrWhiteSpace(val))
+            {
+                return val.Replace(BaseNfoSaver.KodiYouTubeWatchUrl, BaseNfoSaver.YouTubeWatchUrl, StringComparison.OrdinalIgnoreCase);
+            }
+
+            return default;
+        }
+
+        internal static string[] ReadStringArrayFromNfo(this XmlReader reader)
+        {
+            var val = reader.ReadElementContentAsString();
+
+            if (!string.IsNullOrWhiteSpace(val))
+            {
+                return val.Split('/')
+                    .Select(i => i.Trim())
+                    .Where(i => !string.IsNullOrWhiteSpace(i))
+                    .ToArray();
+            }
+
+            return Array.Empty<string>();
+        }
+
+        internal static void ReadUniqueIdFromNfo(this XmlReader reader, BaseItem item)
+        {
+            if (reader.IsEmptyElement)
+            {
+                reader.Read();
+                return;
+            }
+
+            var provider = reader.GetAttribute("type");
+            var id = reader.ReadElementContentAsString();
+            if (!string.IsNullOrWhiteSpace(provider) && !string.IsNullOrWhiteSpace(id))
+            {
+                item.SetProviderId(provider, id);
+            }
```

I take it from this you are parsing nicely and ignoring errors in the nfo? 

Just thinking about debugging / tracing issues - would it help to have some sort of logging?

---

Yes I thought about adding logging if the value is invalid. Should it be a warning, info, or debug log?

---

i'd personally do debugs.